apm-lambda-extension/extension/apm_server.go

```diff
@@ -75,17 +91,61 @@ func PostToApmServer(client *http.Client, agentData AgentData, config *extension
 
 	resp, err := client.Do(req)
 	if err != nil {
+		enterBackoff()
 		return fmt.Errorf("failed to post to APM server: %v", err)
 	}
 
 	//Read the response body
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
+		enterBackoff()
 		return fmt.Errorf("failed to read the response body after posting to the APM server")
 	}
 
 	log.Printf("APM server response body: %v\n", string(body))
 	log.Printf("APM server response status code: %v\n", resp.StatusCode)
+	apmServerTransportFailing = false
 	return nil
 }
+
+func EnqueueAPMData(agentDataChannel chan AgentData, agentData AgentData) {
+	select {
+	case agentDataChannel <- agentData:
+		log.Println("Adding agent data to buffer to be sent to apm server")
+	default:
+		log.Println("Channel full: dropping event")
+	}
+}
+
+func IsTransportStatusHealthy() bool {
+	return apmServerTransportFailing != true
```

Comment:
```suggestion
	return !apmServerTransportFailing
```

Reply:
After writing some tests, I realised that an intermediate `waitingForNextAttempt` status is required, in order to avoid a lockdown situation where we do not differentiate between the situation where we enter the backoff from a healthy situation, and the situation where we enter the backoff from an already failing situation.

The `ApmServerTransportStatusType` is re-introduced in the next commits